src/rastervision/backend/tf_object_detection_config.py

```diff
@@ -0,0 +1,352 @@
+
+        if command_type in [rv.PREDICT, rv.BUNDLE]:
+            if not conf.model_uri:
+                io_def.add_missing('Missing model_uri.')
+            else:
+                io_def.add_input(conf.model_uri)
+
+        return (conf, io_def)
+
+
+class TFObjectDetectionConfigBuilder(BackendConfigBuilder):
+    def __init__(self, prev=None):
+        config = {}
+        if prev:
+            config = {
+                'tfod_config': prev.tfod_config,
+                'pretrained_model_uri': prev.pretrained_model_uri,
+                'train_options': prev.train_options,
+                'script_locations': prev.script_locations,
+                'debug': prev.debug,
+                'training_data_uri': prev.training_data_uri,
+                'training_output_uri': prev.training_output_uri,
+                'model_uri': prev.model_uri
+            }
+        super().__init__(rv.TF_OBJECT_DETECTION, TFObjectDetectionConfig,
+                         config, prev)
+        self.config_mods = []
+        self.require_task = prev is None
+
+    def from_proto(self, msg):
+        b = super().from_proto(msg)
+        conf = msg.tf_object_detection_config
+        # Since this is coming from a serialized message,
+        # assume the task has already been set and do not
+        # require it during validation.
+        b.require_task = False
+        if self.config.get('pretrained_model_uri'):
+            b = b.with_pretrained_model_uri(self.config.pretrained_model_uri)
+        b = b.with_train_options(
+            sync_interval=conf.sync_interval, do_monitoring=conf.do_monitoring)
+        b = b.with_script_locations(
+            train_uri=conf.train_py,
+            eval_uri=conf.eval_py,
+            export_uri=conf.export_py)
+        b = b.with_training_data_uri(conf.training_data_uri)
+        b = b.with_training_output_uri(conf.training_output_uri)
+        b = b.with_model_uri(conf.model_uri)
+
+        # TODO: Debug
+        # b = b.with_debug(conf.debug)
+
+        return b.with_template(json_format.MessageToDict(conf.tfod_config))
+
+    def validate(self):
+        super().validate()
+        if not self.config.get('tfod_config'):
+            raise rv.ConfigError('You must specify a template for the backend '
+                                 "configuration - use 'with_template'.")
+        if self.require_task and not self.task:
+            raise rv.ConfigError('You must specify the task this backend '
+                                 "is for - use 'with_task'.")
+        return True
+
+    def build(self):
+        """Build this configuration, setting any values into the
+           TF object detection pipeline config as necessary.
+        """
+        self.validate()
+
+        b = deepcopy(self)
+
+        # Check if a pretrained model was assigned.
+        pretrained_model = b.config.get('pretrained_model_uri')
+        if pretrained_model:
+            b = b.with_config({'fineTuneCheckpoint': pretrained_model})
```

Comment:
I think that setting this has no effect since the backend overwrites `fine_tune_checkpoint` to the local path of the downloaded pretrained model here https://github.com/azavea/raster-vision/blob/feature/api-refactor/src/rastervision/backend/tf_object_detection.py#L495

But even if that wasn't occurring, this wouldn't work because TFOD can't handle S3 URIs.

Reply:
Nevermind this comment. I see that this works now.